File: frontend/web/services/services/areas.ts
import { axiosInstance } from "@/services";
import { SERVICE_ERROR_UNKNOWN } from "@/config/services";
import { ServiceReturn } from "@/types/api";
import { Area } from "@/types/services";

const getServiceActions = async (serviceId: string): Promise<ServiceReturn<Area[]>> => {
	try {
		const response = await axiosInstance.get<Area[]>(`/services/${serviceId}/actions`);
		return { data: response.data, error: undefined };
	} catch (error) {
		return { data: null, error: SERVICE_ERROR_UNKNOWN };
	}
};

const getServiceReactions = async (serviceId: string): Promise<ServiceReturn<Area[]>> => {
	try {
		const response = await axiosInstance.get<Area[]>(`/services/${serviceId}/reactions`);
		return { data: response.data, error: undefined };
	} catch (error) {
		return { data: null, error: SERVICE_ERROR_UNKNOWN };
	}
};

export { getServiceActions, getServiceReactions };

File: frontend/mobile/lib/services/services/areas.dart
import 'package:area_mobile/services/dio.dart';
import 'package:area_mobile/types/services.dart';

Future<ServiceReturn<List<Area>>> getServiceActions(String serviceId) async {
  try {
    final response =
        await dio.get<List<dynamic>>('/services/$serviceId/actions');
    return ServiceReturn(
        data: response.data!.map((e) => Area.fromJson(e)).toList());
  } catch (e) {
    return ServiceReturn(error: e.toString());
  }
}

Future<ServiceReturn<List<Area>>> getServiceReactions(String serviceId) async {
  try {
    final response =
        await dio.get<List<dynamic>>('/services/$serviceId/reactions');

    if (response.data != null) {
      return ServiceReturn(
        data: response.data!.map((e) => Area.fromJson(e)).toList(),
      );
    } else {
      return const ServiceReturn(error: "Response data is null");
    }
  } catch (e) {
    return ServiceReturn(error: e.toString());
  }
}
